nav2_gps_waypoint_follower_demo/models/turtlebot_with_gps/model.sdf

```diff
@@ -0,0 +1,498 @@
+<?xml version="1.0" ?>
+<sdf version="1.5">
+  <model name="turtlebot_with_gps">
```

Comment:
Am I correct in saying this is the same as the TB model we have in Nav2 + a GPS sensor? If not, please change so it is :-)

Reply:
You are correct, its [this](https://github.com/ROBOTIS-GIT/turtlebot3_simulations/tree/ros2/turtlebot3_gazebo/models/turtlebot3_waffle) model with a GPS sensor. 

Is there any way I could re-use the existing model and just add the GPS? also I will rename it to `turtlebot_waffle_with_gps`

Reply:
Good question, I suspect... maybe? Can you import other models into your model in URDFs? Its been a few years since I've had to make a bunch of URDFs and that's a skill that's use-it-or-lose-it. I don't mind too much for the copy/paste here though, but I'd spend 10 minutes looking into it, no more. 

Don't use the TB3 model that you linked to though. Use ours in Nav2.

Reply:
I reviewed this and it's indeed possible to reuse nav2's turtlebot model, and just add a gps sensor:
```.xml
<?xml version="1.0" ?>
<sdf version="1.5">
  <model name="turtlebot_waffle_gps">
      <include>
        <uri>model://turtlebot3_waffle</uri>
      </include>
      <pose>0.0 0.0 0.0 0.0 0.0 0.0</pose>

      <link name="gps_link">
        <inertial>
          <pose>-0.052 0 0.111 0 0 0</pose>
           ...
        </inertial>
        <sensor name="tb3_gps" type="gps">
          <always_on>true</always_on>
          <update_rate>1</update_rate>
          <pose>0 0 0 0 0 0</pose>
          <gps>
             ...
          </gps>
          <plugin name="my_gps_plugin" filename="libgazebo_ros_gps_sensor.so">
            <ros>
              <remapping>~/out:=/gps/fix</remapping>
            </ros>
          </plugin>
        </sensor>
      </link>
    </model>

</sdf>
```

However, when I do this the frame ids of all the data coming from the plugins inside the included `turtlebot3_waffle`  get the `turtlebot3_waffle` prefix appended at the beginning (ex: `base_scan` -> `turtlebot3_waffle::base_scan`), which then nodes unable to transform messages unless links are artificially renamed. 

Have you encountered this issue before?

Reply:
Sorry, I think lack of clarity on my side is to blame. I meant copy+paste the Nav2 model and add a GPS sensor here. There are modifications of the Nav2 version which are important over the TB3 supported version (which they haven't been very attentive maintainers to actually, you know, address, so we had to fork it). 

You're also welcome to use another robot model as you like. I don't know if you have a kiwi model, but that could be cool! If nothing else, for the video, even if the tutorial code has the TB3 files

Reply:
It's now the same model as nav2's with:
1. a gps sensor
2. the diff drive plugin not publishing the `odom`->`base_footprint` tf, this is published by one ekf in RL

Reply:
I think there's a small error in nav2's turtlebot model file: the collision and pose tags in the realsense camera are duplicated (collision in [L366](https://github.com/ros-planning/navigation2/blob/main/nav2_bringup/worlds/waffle.model#L366) and [L394](https://github.com/ros-planning/navigation2/blob/main/nav2_bringup/worlds/waffle.model#L394) and pose in [L355](https://github.com/ros-planning/navigation2/blob/main/nav2_bringup/worlds/waffle.model#L355) and [L402](https://github.com/ros-planning/navigation2/blob/main/nav2_bringup/worlds/waffle.model#L402)). I removed the duplicated tags in my model, should I do the same for nav2's?

Reply:
Sure!